plugins/UM3NetworkPrinting/src/util.py

```diff
@@ -0,0 +1,37 @@
+
+from typing import Any, Dict, Optional
+
+from UM.Logger import Logger
+
+
+# Finds all machines (GlobalStacks) with the given network_key or group_name and update the metadata of those
+# machines with the key/values in the given metadata dict.
+# Note that the old authentication data "network_authentication_id" and "network_authentication_key" will be
+# deleted if found.
+def updateMetadataForNetworkGroupMachines(network_key: Optional[str], group_name: Optional[str],
+                                          new_metadata: Dict[str, Any]) -> None:
+    # Either match with network_key or group_name
+    filter_dict = {}
+    if network_key is not None:
+        filter_dict["um_network_key"] = network_key
+    if group_name is not None:
+        filter_dict["connect_group_name"] = group_name
+
+    from cura.CuraApplication import CuraApplication
```

Comment:
Why is this import not at the top of the file? I would hope that there is not a circular dependency on this because this is in a plug-in, and CuraApplication should not import plug-ins.